docs/articles/documentation/using-testcafe/common-concepts/inject-scripts-into-tested-pages.md

```diff
@@ -0,0 +1,445 @@
+
+    ```json
+    {
+        "clientScripts": ["mockDate.js", "scripts/react-helpers.js"]
+    }
+    ```
+
+## Add Client Scripts to Specific Tests
+
+Use the [fixture.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) and [test.clientScripts](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages) methods to inject scripts into pages visited during a particular test or fixture.
+
+```js
+fixture `My fixture`
+    .page `http://example.com`
+    .clientScripts('assets/jquery.js');
+```
+
+```js
+test
+    ('My test', async t => { /* ... */ })
+    .clientScripts({ module: 'async' });
+```
+
+## Provide Scripts to Inject
+
+You can pass the following arguments to specify the scripts to inject:
+
+* [path to a JavaScript file](#inject-a-javascript-file)
+* [module name](#inject-a-module)
+* [script code](#inject-script-code)
+
+You can also [inject scripts into specific pages](#provide-scripts-for-specific-pages).
+
+> Note that the API methods and configuration option support [multiple arguments](#specify-multiple-scripts).
+
+### Inject a JavaScript File
+
+Specify the JavaScript file path to inject the entire content of this file into the tested pages. You can pass a string or an object with the `path` property.
+
+*CLI*
+
+```text
+testcafe <browser> <tests> --cs <filePath>[,<filePath2>,...<filePathN>]
+```
+
+*Programming interface*
+
+```text
+runner.clientScripts(filePath | { path: filePath })
+runner.clientScripts(filePath | { path: filePath }, ...)
+runner.clientScripts([ filePath | { path: filePath } ])
+```
+
+*Configuration file*
+
+```text
+{
+    "clientScripts": "<filePath>" | { "path": "<filePath>" }
+}
+{
+    "clientScripts": [ "<filePath>" | { "path": "<filePath>" } ]
+}
+```
+
+*Test API*
+
+```text
+fixture.clientScripts(filePath | { path: filePath })
+fixture.clientScripts(filePath | { path: filePath }, ...)
+fixture.clientScripts([ filePath | { path: filePath } ])
+```
+
+```text
+test.clientScripts(filePath | { path: filePath })
+test.clientScripts(filePath | { path: filePath }, ...)
+test.clientScripts([ filePath | { path: filePath } ])
+```
+
+Argument   | Type   | Description
+---------- | ------ | ---------------------------------------------------------------------------
+`filePath`, `filePath2`, `filePathN` | String | The path to the JavaScript file whose content should be injected.
+
+#### Relative Paths
+
+Relative paths resolve from the *current working directory* when you inject scripts in:
+
+* [command line interface](../command-line-interface.md#--cs-pathpath2---client-scripts-pathpath2)
+* [programming interface](../programming-interface/runner.md#clientscripts)
+* [configuration file](../configuration-file.md#clientscripts)
+
+When you use [test API methods](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages), relative paths resolve from the *test file location*.
```

Comment:
When you use [test API methods](../../test-api/test-code-structure.md#inject-scripts-into-tested-pages), relative paths are resolved from the *test file location*